fetch-in-react/my-app/src/UserDetails.tsx

```diff
@@ -0,0 +1,55 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { useEffect, useState } from 'react';
+import { UserCard } from './UserCard';
+import type { User } from './Users';
+
+type Props = {
+  userId: number;
+  onCancel: () => void;
+};
+
+export function UserDetails({ userId, onCancel }: Props) {
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<unknown>();
+  const [user, setUser] = useState<User>();
+
+  /* your code here (hint: useEffect) */
+
+  useEffect(() => {
+    async function getUserDetails() {
+      console.log('getUserDetails');
+      try {
+        const res = await fetch(
+          `https://jsonplaceholder.typicode.com/users/${userId}`
+        );
+        if (!res.ok)
+          throw new Error(`Get users error: ${res.status} ${res.statusText}`);
+        const responseUser = await res.json();
+        if (!responseUser) throw new Error('failed to get users');
```

Comment:
I'm not totally sure this would be needed. If a user wasn't found there would be a 404 response meaning that res.ok on line 25 would be false so your earlier if statement should be enough to handle this properly.

Reply:
oh ok, figured the same thing but I wasn't sure. Added it just to be safe